indico/modules/legal/forms.py

```diff
@@ -5,14 +5,20 @@
 # modify it under the terms of the MIT License; see the
 # LICENSE file for more details.
 
-from wtforms.fields import TextAreaField, URLField
-from wtforms.validators import URL, Optional
+from wtforms.fields import BooleanField, TextAreaField, URLField
+from wtforms.validators import URL, InputRequired, Optional
 
 from indico.util.i18n import _
 from indico.web.forms.base import IndicoForm
+from indico.web.forms.fields import IndicoDateField
 from indico.web.forms.widgets import TinyMCEWidget
 
 
+class AgreementForm(IndicoForm):
+    # Label needs to be set in the html template
+    accept_terms = BooleanField(_('Accept terms'), [InputRequired()])
```

Comment:
The change you made to add the label here might make it a bit ambiguous. The checkbox says accept terms, but the descriptions says to accept the terms and privacy policy. I had left the label empty here so that the description is the label, and the checkbox specifically says accept the terms and privacy policy.

Reply:
Empty label does not look great though, and more importantly you lose the ability to click something that toggles the checkbox (which also makes this an a11y issue).

I wouldn't mind setting the label dynamically, but modifying the label on the fly is a bit ugly. The cleanest way is probably to simply create the form itself dynamically (ie create the class on the fly - there are a few examples in the codebase where we use `form_cls = type('_SomeForm', (IndicoForm,), {...})`) so you can set the label dynamically as well.

Anyway, the idea was that "terms" is generic and thus covers any kind of terms you have to accept, and then the description shows the verbose details on what you're accepting.

An alternative could be adding two checkboxes (both required ofc) if there are ToS and Privacy Policy to accept. That avoids all the not-so-pretty dynamic generation of the description as well... Code-wise that's also be super easy: Have both checkboxes in the form class, and use `del self.whatever` to remove the ones not needed in the form's `__init__` after the super call.

Reply:
I think I'd like to go with setting the label dynamically and have done so with a dynamic form. I've run in to an issue where the text wraps into the next line. While I could certainly fix it for English by tweaking the text, it should probably be fixed in HTML/CSS instead. This could affect very many forms though, so I haven't taken the plunge here. If you feel more comfortable changing the styles here I'd much appreciate.

<img width="858" alt="image" src="https://github.com/indico/indico/assets/607198/7f3fe38c-b5ca-4925-b95a-e5e460311b8b">